libraries/Microsoft.Bot.Connector/OAuthClientConfig.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.Bot.Schema;
+using Microsoft.Rest;
+using Newtonsoft.Json;
+
+namespace Microsoft.Bot.Connector
+{
+    public static class OAuthClientConfig
+    {        
+        /// <summary>
+        /// The default endpoint that is used for API requests.
+        /// </summary>
+        public static string OAuthEndpoint { get; set; } = AuthenticationConstants.OAuthUrl;
+        
+        /// <summary>
+        /// When using the Emulator, whether to emulate the OAuthCard behavior or use connected flows
+        /// </summary>
+        public static bool EmulateOAuthCards { get; set; } = false;
+
+        /// <summary>
+        /// Send a dummy OAuth card when the bot is being used on the Emulator for testing without fetching a real token.
+        /// </summary>
+        /// <param name="emulateOAuthCards">Indicates whether the Emulator should emulate the OAuth card.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        public static async Task SendEmulateOAuthCardsAsync(OAuthClient client, bool emulateOAuthCards)
+        {
+            // Tracing
+            bool _shouldTrace = ServiceClientTracing.IsEnabled;
+            string _invocationId = null;
+            if (_shouldTrace)
+            {
+                _invocationId = ServiceClientTracing.NextInvocationId.ToString();
+                Dictionary<string, object> tracingParameters = new Dictionary<string, object>();
+                tracingParameters.Add("emulateOAuthCards", emulateOAuthCards);
+                ServiceClientTracing.Enter(_invocationId, client, "GetToken", tracingParameters);
+            }
+            // Construct URL
+            var _baseUrl = client.BaseUri.AbsoluteUri;
+            var _url = new Uri(new Uri(_baseUrl + (_baseUrl.EndsWith("/") ? "" : "/")), "api/usertoken/emulateOAuthCards?emulate={emulate}").ToString();
+            _url = _url.Replace("{emulate}", emulateOAuthCards.ToString());
+
+            // Create HTTP transport objects
+            var _httpRequest = new HttpRequestMessage();
+            HttpResponseMessage _httpResponse = null;
+            _httpRequest.Method = new HttpMethod("POST");
+            _httpRequest.RequestUri = new System.Uri(_url);
+
+            var cancellationToken = CancellationToken.None;
+
+            // Serialize Request
+            string _requestContent = null;
+            // Set Credentials
+            if (client.Credentials != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This could/would never hit. The variable is set on line 55, and is always "None".